File: app/src/androidTest/java/com/example/collections_and_maps/ui/benchmark/models/AppComponent.java
package com.example.collections_and_maps.ui.benchmark.models;

import com.example.collections_and_maps.ui.benchmark.BenchmarkViewModelFactory;

import dagger.Component;

@Component(modules = {AppModuleTest.class})
public interface AppComponent {
    void inject(BenchmarkViewModelFactory factory);

}

File: app/src/main/java/com/example/collections_and_maps/models/AppComponent.kt
package com.example.collections_and_maps.models

import com.example.collections_and_maps.ui.benchmark.BenchmarkViewModelFactory
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [AppModule::class])
interface AppComponent {
    fun inject(factory: BenchmarkViewModelFactory?)
}
